security/pkg/nodeagent/secrets/secretserver.go

```diff
@@ -50,8 +50,6 @@ func NewSecretServer(cfg *Config) (SecretServer, error) {
 	switch cfg.Mode {
 	case SecretFile:
 		return &SecretFileServer{cfg.SecretDirectory}, nil
-	case SecretDiscoveryServiceAPI:
```

Comment:
What's the overall plan? Should we delete `SecretDiscoveryServiceAPI` definition as well?

Reply:
draft plan is to keep this folder for file storage only, will have a new pkg for memory storage and key rotation refresher job. will remove SecretDiscoveryServiceAPI in later PRs.